core/cloud_connected_optimizer.py

```diff
@@ -0,0 +1,58 @@
+import logging
+import time
+
+from core.utils.enums 					import ActionableEventTypes, RawEventTypes, MessageEventTypes
+from core.utils.enums 					import JobTypes, callback_handler_no_default
+from core.utils.dmljob 					import (DMLJob, DMLAverageJob, DMLCommunicateJob, DMLServerJob, \
+												DMLInitializeJob, DMLSplitJob, DMLTrainJob, DMLValidateJob)
+from core.utils.keras 					import serialize_weights, deserialize_weights
+from core.utils.dmlresult 				import DMLResult
+from core.blockchain.blockchain_utils	import TxEnum, content_to_ipfs
+from core.fed_avg_optimizer import FederatedAveragingOptimizer
+
+
+class CloudConnectedOptimizer(FederatedAveragingOptimizer):
+    """
+    This is a clone of FedAvgOptimizer, but for two differences.
+    1) This Optimizer requires the job_uuid field to be set in
+    its initialization payload
+    2) When this is finished training, it uploads statistics to
+    a centralized cloud repo
```

Comment:
I like the idea of using inheritance here, but I have some thoughts.
1. I disagree that only this class should have the job_uuid field, because every optimizer should need the job_uuid in order to relay information back to Explora, right?
2. I like that posting statistics is sort of separate from the core functionality of the optimizer. But I don't understand why this functionality is in a subclass vs a new component. Is there any reason why we would still use the old optimizer over this one?